realm/realm-library/src/main/cpp/java_accessor.hpp

```diff
@@ -219,6 +323,67 @@ inline JPrimitiveArrayAccessor<jlongArray, jlong>::ElementsHolder::~ElementsHold
     }
 }
 
+template <>
+inline StringData JavaAccessorContext::unbox(util::Any& v, bool, bool) const
+{
+    if (!v.has_value()) {
+        return StringData();
+    }
+    auto& value = any_cast<JStringAccessor&>(v);
+    return value;
+}
+
+template <>
+inline BinaryData JavaAccessorContext::unbox(util::Any& v, bool, bool) const
+{
+    if (!v.has_value())
+        return BinaryData();
+    auto& value = any_cast<JByteArrayAccessor&>(v);
+    return value.transform<BinaryData>();
+}
+
+template <>
+inline Timestamp JavaAccessorContext::unbox(util::Any& v, bool, bool) const
+{
+    return v.has_value() ? from_milliseconds(unbox<jlong>(v)) : Timestamp();
+}
+
+template <>
+inline RowExpr JavaAccessorContext::unbox(util::Any&, bool, bool) const
+{
+    REALM_TERMINATE("not supported");
+}
+
+template <>
+inline util::Optional<bool> JavaAccessorContext::unbox(util::Any& v, bool, bool) const
+{
+    return v.has_value() ? util::make_optional(unbox<jboolean>(v) == JNI_TRUE) : util::none;
```

Comment:
that actually won't compile. since with `util:make_optional(nullptr)` you need to define a template method `realm::util::Optional<std::nullptr_t>`

`util::none` is special  `Optional` type, it means null. And all the templated `Optional` class has a constructor with `None` type. 

See https://github.com/realm/realm-core/blob/master/src/realm/util/optional.hpp#L41
and https://github.com/realm/realm-core/blob/master/src/realm/util/optional.hpp#L173